jmh-core/src/main/java/org/openjdk/jmh/generators/core/StateObjectHandler.java

```diff
@@ -792,12 +777,7 @@ public boolean hasInvocationStubs(MethodInfo method) {
                     result.add("            f.set(val, " + so.getParamAccessor(paramField) + ");");
                 }
             }
-            for (HelperMethodInvocation hmi : so.getHelpers()) {
-                if (hmi.helperLevel != Level.Trial) continue;
-                if (hmi.type != HelperType.SETUP) continue;
-                Collection<String> args = so.helperArgs.get(hmi.method.getQualifiedName());
-                result.add("        val." + hmi.method.getName() + "(" + Utils.join(args, ",") + ");");
-            }
+            addStateInitializersHelpers(result, so);
```

Comment:
Shouldn't the block above also be replaced with `addStateInitializersParamsLabels`?

Reply:
Actually there is a lot of duplicated lines across the `generators.core` package but in the same time there is almost no tests. I decided not to touch this code at all. So I rolled back the commit.
It will be very easy to find such duplicates with IntelliJ so nothing really valuable there.